src/Exchange/exg-gwfgwf.f90

```diff
@@ -1771,7 +1771,7 @@ subroutine gwf_gwf_da(this)
     call mem_deallocate(this%condsat)
     call mem_deallocate(this%idxglo)
     call mem_deallocate(this%idxsymglo)
-    call mem_deallocate(this%simvals)
+    call mem_deallocate(this%simvals, 'SIMVALS', this%memoryPath)
```

Comment:
Can you remind me why these particular changes are now necessary?

Reply:
The mem_deallocate method has 2 ways of finding the variable to deallocate. If a name and path is provided it first tries to find the variable using that. If it fails or if no name and path is provided it tries to find the variables by matching the pointer (e.g. the this%simvals with the pointer in a memorytype)

The second thing that is important is that there are 2 types of memory types , a master record and copies.

As mentioned before, when no name and path is provided it tries to find the variable by matching the pointer. When iterating through a regular list it will go through the items in the order they are added. As master items are added before copies this will be matched and returned.

However the iterator of a HashTable doesn't have the same ordering characteristics of a List iterator. This means that a copy can be hit before the master record.  This caused an error because the master record is never deallocated.

A master record and a copy do have different names. Therefor by matching the memory types by name and path will result in the correct memory type to be deallocated